packages/client/hmi-client/src/components/workflow/scenario-templates/value-of-information/value-of-information-scenario.ts

```diff
@@ -0,0 +1,317 @@
+			!!this.modelSpec.id &&
+			!!this.modelConfigSpec.id &&
+			!this.interventionSpecs.some((interventionSpec) => !interventionSpec.id) &&
+			!this.parameters.some((parameter) => !parameter) &&
+			!_.isEmpty(this.simulateSpec.ids)
+		);
+	}
+
+	async createWorkflow() {
+		const wf = new workflowService.WorkflowWrapper();
+		wf.setWorkflowName(this.workflowName);
+		wf.setWorkflowScenario(this.toJSON());
+
+		// 1. Add model and compare dataset nodes
+		const modelNode = wf.addNode(
+			ModelOp,
+			{ x: 0, y: 0 },
+			{
+				size: OperatorNodeSize.medium
+			}
+		);
+
+		wf.updateNode(modelNode, {
+			state: {
+				modelId: this.modelSpec.id
+			},
+			output: {
+				value: [this.modelSpec.id]
+			}
+		});
+
+		const compareDatasetNode = wf.addNode(
+			CompareDatasetOp,
+			{ x: 0, y: 0 },
+			{
+				size: OperatorNodeSize.medium
+			}
+		);
+
+		// add input ports for each simulation to the dataset transformer, this will be a matrix of intervention x parameter
+		for (let i = 0; i < this.interventionSpecs.length * this.parameters.length; i++) {
+			workflowService.appendInputPort(compareDatasetNode, {
+				type: 'datasetId|simulationId',
+				label: 'Dataset or Simulation'
+			});
+		}
+
+		let compareDatasetIndex = 0;
+
+		const modelConfig = await getModelConfigurationById(this.modelConfigSpec.id);
+
+		// chart settings for simulate node
+		let simulateChartSettings: ChartSetting[] = [];
+		simulateChartSettings = updateChartSettingsBySelectedVariables(
+			simulateChartSettings,
+			ChartSettingType.VARIABLE,
+			this.simulateSpec.ids
+		);
+
+		// 2. create model config nodes for each paramter and attach them to the model node
+		const modelConfigPromises = this.parameters.flatMap(async (parameter) => {
+			if (!parameter) return null;
+			const clonedModelConfig = _.cloneDeep(modelConfig);
+
+			const modelConfigNode = wf.addNode(
+				ModelConfigOp,
+				{ x: 0, y: 0 },
+				{
+					size: OperatorNodeSize.medium
+				}
+			);
+
+			wf.addEdge(modelNode.id, modelNode.outputs[0].id, modelConfigNode.id, modelConfigNode.inputs[0].id, [
+				{ x: 0, y: 0 },
+				{ x: 0, y: 0 }
+			]);
+
+			setParameterDistribution(clonedModelConfig, parameter.referenceId, parameter.distribution);
+
+			clonedModelConfig.name = `${modelConfig.name}_${parameter.referenceId}`;
+			const newModelConfig = await createModelConfiguration(clonedModelConfig);
+			await useProjects().addAsset(
+				AssetType.ModelConfiguration,
+				newModelConfig.id,
+				useProjects().activeProject.value?.id
+			);
+
+			wf.updateNode(modelConfigNode, {
+				state: {
+					transientModelConfig: newModelConfig
+				},
+				output: {
+					value: [newModelConfig.id],
+					state: _.omit(modelConfigNode.state, ['transientModelConfig'])
+				}
+			});
+
+			return modelConfigNode;
+		});
+
+		// Wait for all modelConfigPromises to resolve and filter non null responses
+		const modelConfigNodes = (await Promise.all(modelConfigPromises)).filter((node) => !!node);
```

Comment:
what is with the filter at the end? what does it do?

Reply:
Since there's an early return in the `flatMap` function when it can't find a parameter.  The return of the promise can be a `WorkflowNode<any>` or `null`.  The filter will filter out all the promises that return `null`.

Reply:
So I am not sure why you need to use `flatMap`, the structure is only a single level no?

But that aside, if you know that result cannot be null, just declare or coerce the data as such or sprinkle in `@ts-ignore`. We are not obligated to satisfy Typescript if it makes non-sensical warnings. Using additional control structure not only make reading more confusing, it also introduce additional performance cost with no clear benefits.